app/services/catalog/create_approval_request.rb

```diff
@@ -0,0 +1,66 @@
+require 'approval_api_client'
+
+module Catalog
+  class CreateApprovalRequest
+    attr_reader :order
+
+    def initialize(id)
+      @order = Order.find_by!(:id => id)
+
+      ApprovalApiClient.configure do |config|
+        config.api_key['x-rh-auth-identity'] = ManageIQ::API::Common::Request.current.request_id
+      end
+
+      # Set up the instance class-wide since there are N portfolio_items and workflows to request
+      @api_instance = ApprovalApiClient::RequestApi.new
+    end
+
+    def process
+      @order.order_items.each do |item|
+        approvals = send_approval_requests(item)
+
+        approvals.compact.each do |req|
+          @order.approval_requests << create_approval_request(req)
+        end
+      end
+
+      @order.update(:status => "Ordered", :ordered_at => Time.now.utc)
+
+      self
+    rescue ApprovalAPIClient::ApiError => e
+      Rails.logger.error("Error putting in approval Request for #{order.id}: #{e.message}")
+      raise
+    end
+
+    private
+
+    def send_approval_requests(item)
+      [].tap do |requests|
+        workflows(item.portfolio_item).each do |workflow|
```

Comment:
can use `workflows(item.portfolio_item).each_with_object([])`

Reply:
actually shouldn't you use `workflows(item.portfolio_item).collect`?

Reply:
Yep you're right, I got overzealous with `.tap`, I changed it to a `.map`